src/main.js

```diff
@@ -8,22 +8,37 @@ import { sync } from 'vuex-router-sync'
 
 import App from './App'
 import router from './router'
-import store from './store'
-
+import { store } from './store'
 import urls from './urls'
+import { config } from './config'
+
+import * as firebase from 'firebase'
+import firebaseui from 'firebaseui'
 
 sync(store, router)
 Vue.use(Vuetify)
 Vue.use(VueHead)
 Vue.config.productionTip = false
 
 Vue.prototype.$urls = urls
-
+Vue.prototype.$firebase = firebase.initializeApp(config.firebase)
+Vue.prototype.$auth = firebase.auth()
+Vue.prototype.$authUI = new firebaseui.auth.AuthUI(Vue.prototype.$auth)
```

Comment:
Let's not add these stuff in the global namespace. Because they will not be used directly in the UI but through store or some component

Reply:
We are using this.$authUI in Signin.vue

Reply:
Yes

Reply:
Its asking me to reinitialise the app (even though the app is initialised in main.js) 🤔

Reply:
Don't initialize in main.js. Initialize here only. Just don't store in Vue instance

You can store firebase instance in Vue but not auth and authUI as they are derivable

Bloating the instance is a bad practice

Reply:
It was asking to initialise the app because the JS of signin.vue was loading before the JS of main.js and there was no firebase app initialised at that time (asynchronous nature :question: ) . I created the firebaseUI instance inside `mounted ()` and its working now, even when the firebase app is initialised in main.js. I think its better to initialise it in main.js rather than in some secondary module.

Reply:
@iamareebjamal  Should I put the `firebase.initializeApp()` back into main.js ?